File: python/day77.py
import getpass
import string

# 1. Securely ask for password input (hides character typing)
password = getpass.getpass("Enter your password: ")

# 2. Define your symbols (string.punctuation includes !"#$%&'()*+,-./:;<=>?@[\]^_`{|}~)
symbols = set(string.punctuation)

# 3. Check if at least one character is a symbol
has_symbol = any(char in symbols for char in password)

if has_symbol:
    print("Success: Password contains a symbol.")
else:
    print("Error: Password must contain at least one symbol.")

File: js/day77.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// 1. Ask for password input
rl.question('Enter your password: ', (password) => {
    
    let hasSymbol = false;

    // 2. Loop through character codes
    for (let i = 0; i < password.length; i++) {
        const code = password.charCodeAt(i);
        
        // Exclude numbers (48-57), uppercase (65-90), lowercase (97-122)
        const isNumber = code >= 48 && code <= 57;
        const isUpper = code >= 65 && code <= 90;
        const isLower = code >= 97 && code <= 122;
        
        if (!isNumber && !isUpper && !isLower) {
            hasSymbol = true;
            break; // Stop looping early once a symbol is found
        }
    }

    // 3. Output the result
    if (hasSymbol) {
        console.log('Success: Password contains a symbol.');
    } else {
        console.log('Error: Password must contain at least one symbol.');
    }

    rl.close();
});
